core/src/main/scala/com/spotify/featran/transformers/IQROutlierRejector.scala

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2017 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.featran.transformers
+
+import com.spotify.featran.FeatureRejection
+import com.twitter.algebird.QTreeAggregator
+
+/**
+  * Transform features by passing them through, and rejecting them if they
+  * fall outside of either 1.5 times the IQR below the first quartile,
+  * or 1.5 times the IQR above the third.
+  *
+  * The IGR or inter quartile range is the range between the first and the third quartiles.
+  *
+  * The bin ranges are chosen using the Algebird's QTree approximate data structure. The precision
+  * of the approximation can be controlled with the `k` parameter.
+  *
+  * Missing values are transformed to zero vectors.
+  *
+  * Values contained in the last bucket will report [[FeatureRejection.Outlier]] as rejection.
+  *
+  * When using aggregated feature summary from a previous session, values outside of previously seen
+  * `[min, max]` will also report [[FeatureRejection.Outlier]] as rejection.
+  */
+object IQROutlierRejector {
+
+  /**
+    * Create a new [[IQROutlierRejector]] instance.
+    *
+    * @param rejectLower whether to reject outlier based on the lower igr bound.
+    * @param rejectUpper whether to reject outlier based on the upper iqr bound.
+    * @param k           precision of the underlying Algebird QTree approximation
+    */
+  def apply(name: String,
+            rejectLower: Boolean = true,
+            rejectUpper: Boolean = true,
+            k: Int = QTreeAggregator.DefaultK)
+  : Transformer[Double, BaseQuantileRejector.B, BaseQuantileRejector.C] =
+    new IQROutlierRejector(name, rejectLower, rejectUpper, k)
+
+}
+
+private class IQROutlierRejector(name: String,
+                          override val rejectLower: Boolean,
```

Comment:
Indentation. Also since you're passing these params to super constructor, you can remove the `override val`.

Reply:
This transformer is bypassed if both `rejectLower` & `rejectUpper` are false right? Should we add a check for that?

Reply:
it is checked in `QuantileOutlierRejector`